File: 20110727abbr/abbr.pas
const spe:array[1..10]of string=('i','pa','te','ni','niti','a','ali','nego','no','ili');
var ts,ss:string;
    t,cnt,i:longint;
    pd:boolean;
begin
assign(input,'abbr.in');
reset(input);
assign(output,'abbr.out');
rewrite(output);
readln(ts);
t:=pos(' ',ts);
cnt:=0;
while t<>0 do
 begin
 ss:=copy(ts,1,t-1);
 delete(ts,1,t);
 pd:=true;
 for i:=1 to 10 do
     if ss=spe[i] then begin pd:=false; break; end;
 inc(cnt);
 if (pd)or(cnt=1) then write(char(64+ord(ss[1])-96));
 t:=pos(' ',ts);
 end;
pd:=true;
for i:=1 to 10 do
    if ts=spe[i] then begin pd:=false; break; end;
inc(cnt);
if (pd) then write(char(64+ord(ts[1])-96));
writeln;
close(input);
close(output);
end.

File: 20110727abbr/abbr.cpp
#include<cstdio>
#include<cstdlib>
#include<cstring>
#include<iostream>
#include<algorithm>
using namespace std;
const char spe[10][6]={"i","pa","te","ni","niti","a","ali","nego","no","ili"};
int cnt,t;
char ts[257],ss[257],*s1;
bool pd;
int main()
{
    freopen("abbr.in","r",stdin);
    freopen("abbr.out","w",stdout);
    scanf("%[^\n]",ts);
    s1=ts;
    t=strcspn(s1," ");
    cnt=0;
    while (t!=strlen(s1))
    {
          strncpy(ss,s1,t);
          strcpy(s1,s1+t+1);
          pd=true;
          for (int i=0;i<10;i++) if (t==strlen(spe[i]) && strncmp(ss,spe[i],t)==0) { pd=false; break; }
          cnt++;
          if (pd || cnt==1) printf("%c",(char)(ss[0]-'a'+'A'));
          t=strcspn(s1," ");
    }
    pd=true;
    for (int i=0;i<10;i++)
        if (strcmp(s1,spe[i])==0) { pd=false; break; }
    cnt++;
    if (pd) printf("%c",(char)(s1[0]-'a'+'A'));
    return 0;
}
